### DISCUSSION TITLE
Disable button

### DISCUSSION THREAD

**Discussion by LuckyTurtleDev** [hecrj/iced#1165] [Q&A] [answered]
How can I disable a button after I had add a on_press event?
For example I would like to enable the button only, if two TextInputs are filled out.

**Comment by alex13sh** [comment, marked as answer]
Just add a condition into the update function. 
For visual display use style

**Reply by LuckyTurtleDev** [reply]
Thanks.
I had hope that an option exist to set a disable attribute.

**Reply by 13r0ck** [reply]
Because Iced is inspired by a functional language, you can also do it the functional way.

Make a helper function that returns the button you want, then in your code have a match statement that returns either the button with an on_press or not

```
match case {
    Ok(_) =>
        button_helper().on_press()
    Err(_) =>
        button_helper()
}

~~~

button_helper() -> {
    Button::new()
}

```
